### ISSUE TITLE
Ideas - Increase perfomance on low/mid devices.

### ISSUE THREAD

**Issue by mayconmesquita** [Flipkart/recyclerlistview#415]
Hi,

I'm using this project so much in my apps. This solves a HUGE problem in RN (recyclerlist).

However, the blank space while scrolling is a fact (discussed here: #41). I know, the RN has an async nature that makes this hard, but impossible?

I think that with help of Swift and Java we can empower that! Maybe using a separate thread like _RN Core Image_ does?

I created this issue to collect ideas that could improve the performance and kill the blank spaces of here!

**Comment by uday5162** [comment]
Hi,

I have also used this in my project and faced the same problem in the dev mode(haven't tried in prod yet). Have you tried setting  renderAheadDistance prop value more than 250 as mentioned in the discussion.

**Comment by naqvitalha** [comment]
There should be any blanks in Snapdragon 625 and above. Can you mention some of the devices where you get them?

**Comment by mayconmesquita** [comment]
@uday5162 I always consider in production env while testing the list, because in dev the . I tried to set from _10_ to _1500_ in _renderAheadDistance_, but _200_ (i think is default) gave the best performance possible. But yes, with blank spaces (high-end devices less, of course).

@naqvitalha I tested the My apps and Flipkart in three devices from the office: Redmi Note 7 (mid-end, Snapdragon 660), Samsung S9 Plus (maybe mid-high end?) and Moto X (low end, Android 5).

If you scroll the _Flipkart Ideas_ fast, the blank space will come. My app has a feed much similar, but with no stories, no videos, only text and image, but the blank space come.

I think that we can try something (experimental), to achieve this.
I realized that Image (or FastImage) increase the blank space possibility, even if using a 1x1 pixel image.

**Comment by naqvitalha** [comment]
@mayconmesquita Did you do any profiling to identify the bottleneck?

**Comment by mayconmesquita** [comment]
> @mayconmesquita Did you do any profiling to identify the bottleneck?

Yes. The UI thread is giving 55-60 fps all time. When scrolling, the JS thread oscillates between 30 fps, and 60 fps.

One thing helped: move all images from _react-native-fast-image_ to **RN Image**, because this last uses its own thread to render images. Per my tests, fast-images handles via JS Thread. So, the performance increased a bit.

I think that Fabric can help us when released.

**Comment by alessandrofc** [comment]
I was experiencing this problem and I fixed it implementing shouldComponentUpdate in my row. The difference was huge.
